### PR TITLE
Bugfix/missing dependencies

### PR DESCRIPTION
**Pull request by AndyOooh**
## Description
- Added all missing dependencies for hardhat.
- Removed ^from all dependencies in `package.json`
- Manually changed versions on `@ethersproject/abi` and `@ethersproject/providers`.
- Added `yarn.lock` to `.gitignore`

## Motivation and Context
Was unable to run `yarn hardhat` commands before.

## How Has This Been Tested?
Manually tested by deleting `node_modules` and running `yarn install` --> `yearn hardhat` and `yarn hardhat test`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Style (style only changes)
- [ ] Refactor (code that does not add new functionality nor fixes a bug)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,6257 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (evchip) [inline yarn.lock:1]: You added `yarn.lock` to the `gitignore` but you're pushing it up in this commit. Do you want to track this file or not?

Comment (AndyOooh) [reply yarn.lock:1]: Meant to ignore it as a measure of safety as I'm not sure if it's desirable to track it or not. What's your opinion?

Comment (evchip) [reply yarn.lock:1]: You should commit the lock file. In which case we don't include it in gitignore.

**Review by evchip — APPROVED** [review]
LGTM

File: yarn.lock (hunk 1 above)
Comment (AndyOooh) [reply yarn.lock:1]: done.

**Pull request merged**